### PR TITLE
[Editorial] Update the docs based on Copilot review feedback

### PR DESCRIPTION
**Pull request by reyang**

### CODE REVIEW

**Review by cijothomas — APPROVED** [review]
how did you use copilot to review :-?

**Review by rajkumar-rangaraj — APPROVED** [review]

**Review by vishweshbankwar — APPROVED** [review]

File: docs/metrics/README.md [hunk 1]
```diff
@@ -163,9 +163,9 @@ Here is the rule of thumb:
 
 > [!NOTE]
 > When reporting measurements with more than 8 tags, the API allocates memory on
-  the hot-path. You SHOULD try to keep the number of tags less than or equal to 8.
-  If you are exceeding this, check if you can model some of the tags as Resource,
-  as [shown here](#metrics-enrichment).
+  the hot code path. You SHOULD try to keep the number of tags less than or
```
Comment (utpilla) [inline docs/metrics/README.md:166]: It's probably better to keep it as hot-path instead of "hot code path"?

**Review by utpilla — APPROVED** [review]

File: docs/metrics/README.md (hunk 1 above)
Comment (reyang) [reply docs/metrics/README.md:166]: CoPilot recommends against "hot-path", a quick search shows that "hot path" and "hot code path" are widely used. e.g. https://www.hackterms.com/hot%20code

File: docs/logs/README.md [hunk 2]
```diff
@@ -32,7 +32,7 @@ OpenTelemetry .NET:
 :heavy_check_mark: You should use structured logging.
 
 * Structured logging is more efficient than unstructured logging.
-  * Filtering and redaction can happen on invidual key-value pairs instead of
+  * Filtering and redaction can happen on indvidual key-value pairs instead of
```
Comment (CodeBlanch) [inline docs/logs/README.md:35]:
```suggestion
  * Filtering and redaction can happen on individual key-value pairs instead of
```

Comment (reyang) [reply docs/logs/README.md]: This is not a CoPilot mistake at all, I got some weird things happening and I haven't figured out why - during copy/paste, some characters would disappear, e.g. https://github.com/open-telemetry/opentelemetry-dotnet/pull/5284#discussion_r1473469810 (obviously I didn't remove it intentionally)

**Pull request merged**